src/useDebouncedCallback.ts

```diff
@@ -115,9 +104,15 @@ export default function useDebouncedCallback<T extends (...args: any[]) => Retur
     if (shouldInvoke(time)) {
       return trailingEdge(time);
     }
-    // Restart the timer.
-    timerId.current = startTimer(timerExpired, remainingWait(time));
-  }, [remainingWait, shouldInvoke, startTimer, trailingEdge]);
+
+    const timeSinceLastCall = time - lastCallTime.current;
+    const timeSinceLastInvoke = time - lastInvokeTime.current;
+    const timeWaiting = wait - timeSinceLastCall;
+    const remainingWait = maxing ? Math.min(timeWaiting, maxWait - timeSinceLastInvoke) : timeWaiting;
```

Comment:
let's comment these block of code, that it's a remaining wait calculation

Reply:
Done